[WIP] azure terraform

serbrech opened the pull request:
PR to own master for local review
terraform part only

cgwalters committed:
rhcos: Embed full build metadata in binary

This way we avoid needing a service available at least for AWS
installs.  The AMIs now get hardcoded into the binary.

`hack/update-rhcos-bootimage.py` is a small script which accepts
a URL to build metadata and updates our cached version with just
the subset of keys we care about (or potentially care about); e.g.
not the pkgdiff.

From Trevor:

Drop the HTTP stuff in favor of just the local asset or
`OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE`.

The codecs business works around the lack of byte-stream support in
json.load before Python 3.6 [1].

[1]: https://docs.python.org/3/library/json.html#json.load

Co-authored-by: Trevor King <wking@tremily.us>

serbrech committed:
data/data/azure: add azure terraform

awesomenix commented on data/data/azure/vnet/nsg.tf:
```diff
@@ -0,0 +1,69 @@
+  direction                   = "Inbound"
+  access                      = "Allow"
+  protocol                    = "tcp"
+  source_port_range           = "10251"
```
Typically not required, since azure by default allows inbound vnet connectivity

serbrech replied:
right, I forgot to drop this one. the aws version had a lot more rules needed, as they close everything by default.
Will revisit the NSG once we have things up and running.

awesomenix commented on data/data/azure/vnet/internal-lb.tf:
```diff
@@ -0,0 +1,69 @@
+  probe_id = "${azurerm_lb_probe.internal_lb_probe_api_internal.id}"
+}
+
+resource "azurerm_lb_rule" "internal_lb_rule_sint" {
```
Whats this?

serbrech replied:
This is the endpoint that the masters query to get their setup from the bootstrap machine, during ignition.
sint stands for service_internal I think. taken from their aws implementation.
Their is a corresponding probe, and only the bootstrap machine will satisfy it, so the call is effectively only routed to the bootstrap machine.

staebler committed:
upi/vsphere: add dns for control plane and compute hostnames

Add DNS records for the control plane and compute hostnames so that
they are resolvable from within the cluster.

serbrech committed:
fix: ran dep ensure on pkg/terraform/exec

staebler committed:
upi/vsphere: use static ip addresses

Static IP addresses are now working with rhcos.

abhinavdahiya committed:
manifests: add cloudproviderconfig asset

This configuration file is used to configure the Kubernetes cloud provider integration
when using the built-in cloud provider integration or the external cloud controller manager.

The configmap containing cloud provider config must be in `openshift-config` namespace as required by openshift/api PR 245 [1]

[1]: https://github.com/openshift/api/blob/master/config/v1/types_infrastructure.go#L28

dav1x committed:
adding only network module for managing IP addresses

change network to IPAM query and add local-exec provisioner for VM on destroy

update examples file by removing IP address and adding machine_cidr

wking committed:
vendor: Bump stretchr/testify to v1.3.0

Pulling in stretchr/testify@199de5f3a4 (Update the LICENSE to match
the 'standard' MIT license file, 2018-08-16, stretchr/testify#648) to
drop the "Please consider promoting this project..." deviation from
the stock MIT license.

Generated with:

  $ dep ensure -update github.com/stretchr/testify

using:

  $ dep version
  dep:
   version     : v0.5.1
   build date  : 2019-03-20
   git hash    : faa61893
   go version  : go1.10.3
   go compiler : gc
   platform    : linux/amd64
   features    : ImportDuringSolve=false

staebler committed:
upi/vsphere: add vsphere to cloudproviderconfig

Add the config for vsphere to cloudproviderconfig.

Add vsphere cloud creds to cloud-creds-secret.yaml.

Add fields to vsphere platform in installconfig to support cloudproviderconfig.

Set the enable_disk_uuid attribute to true for vsphere VMs as that
is required to ensure the VMDK properly mounts disks.

See https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/existing.html
for details on the vsphere cloud config file.

wking committed:
CHANGELOG: Document changes since 0.15.0

Through 58a27678c0 (Merge pull request #1497 from
vrutkovs/upi-multistage-cli, 2019-03-29).

wking committed:
docs/design/resource_dep: Rebuild without ClusterK8sIO, etc.

Catching up with c734361c0e (Remove cluster-api object as this is not
needed anymore, 2019-03-22, #1449), 1408d8a045 (*: use
kube-etcd-cert-signer release image, 2019-03-27, #1477), and possibly
others.  Generated with:

  $ openshift-install graph | dot -Tsvg >docs/design/resource_dep.svg

using:

  $ dot -V
  dot - graphviz version 2.30.1 (20170916.1124)

wking committed:
hack/build: Update release-pin location to defaultReleaseImageOriginal

Catching up with a31e12fc4c (release: Allow release image to be
directly substituted into binary, 2019-03-15, #1422).

abhinavdahiya committed:
upi/metal: update the instance location and size

Bootstrap node is failing to uncompress the bios image with `t1.small` capacity.
and also packet seems to be failing to deploy servers in SJC1. Using `any` allows terraform to create the servers in any available datacenter.

openshift-merge-robot committed:
Merge pull request #1519 from wking/version-0.16.0

CHANGELOG: Document changes since 0.15.0

openshift-merge-robot committed:
Merge pull request #1472 from serbrech/dep-ensure-terraform-exec

fix: Gopkg.lock after running dep ensure on pkg/terraform/exec

runcom committed:
pkg/asset: fix client cert key name

When working on the Ignition v2s3 move, we noticed that we were
generating a duplicate asset. That one had the same name
"apiserver-proxy". The correct name _should_ be "aggregator-client".

Signed-off-by: Antonio Murdaca <runcom@linux.com>

openshift-merge-robot committed:
Merge pull request #1523 from abhinavdahiya/fix_upi

upi/metal: update the instance location and size

openshift-merge-robot committed:
Merge pull request #1522 from wking/fix-build-script-release-image-pin

hack/build: Update release-pin location to defaultReleaseImageOriginal

cuppett committed:
FIXID BZ1695860: Removing hardcoded names on AWS::ElasticLoadBalancingV2::TargetGroup resources

openshift-merge-robot committed:
Merge pull request #1526 from runcom/fix-asset

pkg/asset: fix client cert key name

abhinavdahiya committed:
images/installer: add matchbox provider to upi image

staebler committed:
asset/metadata: save metadata file in asset state file

The Metadata asset uses a private field for its file. This
prevents the file from being saved when the asset state file
is generated. When `create ignition-configs` is called prior
to `create cluster`, then Metadata asset does not have a
file.

This does not present a problem to the end user yet,
because the Metadata asset, as a read-only asset, is not
consumed during the call to `create cluster`.

staebler committed:
asset/password: KubeadminPassword responsible for the password file

The KubeadminPassword asset owns the kube admin password, so that
asset should own creating the password file.

openshift-merge-robot committed:
Merge pull request #1515 from wking/bump-stretchr-testify-for-straight-license

vendor: Bump stretchr/testify to v1.3.0

openshift-merge-robot committed:
Merge pull request #1527 from cuppett/cuppett/bz1695860

BUG 1695860: upi/aws: Removing hardcoded names on TargetGroup resources

openshift-merge-robot committed:
Merge pull request #1423 from cgwalters/rhcos-pin-meta

rhcos: Embed full build metadata in binary

serbrech committed:
pkg/types: Add required azure types

This adds the azure-specific structures for installer
configuration and management.
The property BaseDomainResourceGroupName in pkg/types/azure/platform
should disapear when the installconfig supports the azure
requirements to identify a dns zone.

staebler committed:
upi/vsphere: simplify cidr_to_ip.sh script

We can use the cidr_to_ip.sh external data source to reserve IP
addresses for each machine individually rather than all of the machines
at once. This greatly simplifies the cidr_to_ip.sh script.

Additionally, update the README and terraform.tfvars.example.

openshift-merge-robot committed:
Merge pull request #1530 from abhinavdahiya/metal-ci

images/installer: add matchbox provider to upi image

openshift-merge-robot committed:
Merge pull request #1516 from staebler/vsphere_cloudproviderconfig

upi/vsphere: add vsphere to cloudproviderconfig

openshift-merge-robot committed:
Merge pull request #1518 from staebler/vsphere_updates

upi/vsphere: multiple terraform updates

openshift-merge-robot committed:
Merge pull request #1538 from staebler/fix_metadata_save

asset/metadata: save metadata file in asset state file

openshift-merge-robot committed:
Merge pull request #1539 from staebler/kubeadmin_password_asset_file

asset/password: KubeadminPassword responsible for the password file

Commit:
Update upi.go

Typo "being" -> "begin"

openshift-merge-robot committed:
Merge pull request #1544 from pdmack/pdmack-patch-1

Update upi.go

eparis committed:
Log InstanceProfile name correct on delete

We were passing a *string, not a string, so we logged garbage.

eparis committed:
pkg/asset: Do not generate Files() slices with nil entries

Most asset Files() functions would return an empty list if they didn't have a
file. A couple would instead return a list of length 1 with a nil entry. This
would cause a panic in the file delete code:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x100c8b5]

goroutine 1 [running]:
github.com/openshift/installer/pkg/asset.DeleteAssetFromDisk(0x6dea6c0, 0xc0f8578, 0x7ffdb1fbb99c, 0x1b, 0xc0f8578, 0x5a13001)
	/home/eparis/go/src/github.com/openshift/installer/pkg/asset/asset.go:69 +0x185
github.com/openshift/installer/pkg/asset/store.(*storeImpl).Destroy(0xc000f9e540, 0x7f42d5057770, 0xc0f8578, 0x7f42d5057770, 0xc0f8578)
	/home/eparis/go/src/github.com/openshift/installer/pkg/asset/store/store.go:103 +0x3ee
main.runDestroyCmd(0x7ffdb1fbb99c, 0x1b, 0xc0003e5ec0, 0xc0009b9c60)
	/home/eparis/go/src/github.com/openshift/installer/cmd/openshift-install/destroy.go:60 +0x171
main.newDestroyClusterCmd.func1(0xc0003a3180, 0xc00066ab30, 0x0, 0x1)
	/home/eparis/go/src/github.com/openshift/installer/cmd/openshift-install/destroy.go:38 +0x77
github.com/openshift/installer/vendor/github.com/spf13/cobra.(*Command).execute(0xc0003a3180, 0xc00066ab00, 0x1, 0x1, 0xc0003a3180, 0xc00066ab00)
	/home/eparis/go/src/github.com/openshift/installer/vendor/github.com/spf13/cobra/command.go:766 +0x2ae
github.com/openshift/installer/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0003a2780, 0xc0009b9ee0, 0x1, 0x1)
	/home/eparis/go/src/github.com/openshift/installer/vendor/github.com/spf13/cobra/command.go:852 +0x2ec
github.com/openshift/installer/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/home/eparis/go/src/github.com/openshift/installer/vendor/github.com/spf13/cobra/command.go:800
main.installerMain()
	/home/eparis/go/src/github.com/openshift/installer/cmd/openshift-install/main.go:57 +0x1f3
main.main()
	/home/eparis/go/src/github.com/openshift/installer/cmd/openshift-install/main.go:40 +0x99

openshift-merge-robot committed:
Merge pull request #1549 from eparis/instance-profile-delete-log

pkg/destroy/aws: Log InstanceProfile name correct on delete

wking committed:
cmd/openshift-install/waitfor: Rename from user-provided-infrastructure

There is no hard line between installer- and user-provided
infrastructure.  Rename these commands to focus on what they'll do
instead of the work-flow into which we expect them to fit.

We're still working out how we can drop the router-CA injection to
avoid 'wait-for cluster-ready' surprising users my editing their
on-disk kubeconfig [1].  But that's mitigated somewhat by the fact
that addRouterCAToClusterCA is idempotent, because AppendCertsFromPEM
wraps AddCert [2] and AddCert checks to avoid duplicate certificates
[3].

[1]: https://github.com/openshift/installer/pull/1541
[2]: https://github.com/golang/go/blob/go1.12/src/crypto/x509/cert_pool.go#L144
[3]: https://github.com/golang/go/blob/go1.12/src/crypto/x509/cert_pool.go#L106-L109

abhinavdahiya committed:
*: push pull-secret to openshift-config namespace

This deprecates the kube-system/coreos-pull-secret in favor of openshift-config/pull-secret

openshift-merge-robot committed:
Merge pull request #1548 from eparis/delete-panic

pkg/asset: Do not generate Files() slices with nil entries

abhinavdahiya committed:
docs/design: update the resource_dep graph

openshift-merge-robot committed:
Merge pull request #1554 from abhinavdahiya/move-pull-secret

Bug 1697644: *: push pull-secret to openshift-config namespace

openshift-merge-robot committed:
Merge pull request #1552 from wking/wait-for-renames

cmd/openshift-install/waitfor: Rename from user-provided-infrastructure

openshift-merge-robot committed:
Merge pull request #1534 from serbrech/azure-pkg-types

pkg/types: Add required azure types

staebler committed:
data/manifests: fill out cloud creds role for vsphere platform

Fill out role-cloud-creds-secret-reader.yaml when the platform is vsphere
so that there is proper rbac to the cloud creds secret.

openshift-merge-robot committed:
Merge pull request #1558 from staebler/vsphere_clouds_creds_rbac

data/manifests: fill out cloud creds role for vsphere platform

serbrech committed:
pkg/asset/azure: add azure tfvars generation

serbrech committed:
pkg/asset/cluster: generate azure tfvars

serbrech committed:
pkg/tfvars/azure: gofmt azure.go
